File: 2015/Day_03/Python/test_part_1.py
from . import part_1


def test_separate_directions():
    assert part_1.separate_directions('>') == ['>']
    assert part_1.separate_directions('^>v<') == ['^', '>', 'v', '<']
    assert part_1.separate_directions('^v^v^v^v^v') == ['^', 'v', '^', 'v', '^', 'v', '^', 'v', '^', 'v']


def test_count_house_visits():
    assert part_1.count_houses_visited(['>']) == 2
    assert part_1.count_houses_visited(['^', '>', 'v', '<']) == 4
    assert part_1.count_houses_visited(['^', 'v', '^', 'v', '^', 'v', '^', 'v', '^', 'v']) == 2

File: 2015/Day_03/Ruby/test_part_1.rb
require_relative "part_1"
require 'test/unit'


class TestSantaRoute < Test::Unit::TestCase
    
    def test_separate_directions
        assert_equal(['>'], separate_directions('>'))
        assert_equal(['^', '>', 'v', '<'], separate_directions("^>v<"))
    end
    
    def test_count_houses_visited
        assert_equal(2, count_houses_visited([">"]))
    end
end
